### PR TITLE
Various fixes and improvements

### PR DESCRIPTION
**Pull request by theseion**
I came across a couple of issues while working on 932250 et. al. Hope it's not too much in for one PR.

### CODE REVIEW

**Comment by fzipi** [comment]
Looks like we have failing tests...

**Comment by theseion** [comment]
Fixed.

File: regex/parser/parser.go [hunk 1]
```diff
@@ -247,18 +251,18 @@ func parseFile(rootParser *Parser, filename string, definitions map[string]strin
 // We removed flag merging because of https://github.com/coreruleset/crs-toolchain/issues/72
 func mergePrefixesSuffixes(target *Parser, source *Parser, out *bytes.Buffer) (*bytes.Buffer, error) {
 	logger.Trace().Msg("merging prefixes, suffixes from included file")
+	// If the included file has flags, this is an error
+	if len(source.Flags) > 0 {
+		return out, errors.New("include files must not contain flags. See https://github.com/coreruleset/crs-toolchain/issues/71")
+	}
```
Comment (fzipi) [inline regex/parser/parser.go:257]: I think you are changing the semantics I've intended. Returning out instead of an empty newOut.

**Review by fzipi — COMMENTED** [review]
Made a small comment.

**Pull request merged**